moto_week3/s3_functions.py

```diff
@@ -0,0 +1,43 @@
+import boto3
+
+
+def get_client():
+    return boto3.client("s3")
+
+
+def list_s3_buckets():
+    s3 = get_client()
+    response = s3.list_buckets()
+    for bucket in response['Buckets']:
+        yield(bucket['Name'])
+
+
+def list_s3_objects(bucket):
+    s3 = get_client()
+    response = s3.list_objects_v2(Bucket=bucket)
+    if response:
+        try:
+            for _object in response['Contents']:
+                yield(_object['Key'])
+        except KeyError:
+            pass
+
+
+def read_s3_object(bucket, key):
+    s3 = get_client()
+
+    response = s3.get_object(Bucket=bucket, Key=key)
+    if response:
+
+        yield(response['Body'].read())

```

Comment:
Good practice to maybe return decoded response object.
```suggestion
        yield(response['Body'].read().decode())
```

Reply:
any specific reasons? just curious.

Reply:
It's better to deal with string like objects instead of byte like object. And doing that conversion at the first point of contact would be the most convenient way.

Reply:
yes. that is why i was stuck later when i was dealing with byte like objects, and couldn't find a workaround. thanks a lot.